robot:
  max_dist_tran: 5 # mm, maximum distance between set point and current pose.
  max_dist_rot: 0.05 # rad, maximum distance between set point and current pose.
  safety_mode: 1 # 0: none 1: truncate 2: stop
  operation_mode: 0 # 0: cartesian 1: joint
  safe_zone:
    xmin: -50
    xmax: 500
    ymin: -690
    ymax: -180
    zmin: 0
    zmax: 500


delta:
  base_center2edge: 0.040 # base, distance from center to motor axis
  platform_center2edge: 0.02 # platform, distance from center to axis
  upper_leg_length: 0.1 # upper legs length
  lower_leg_length: 0.1 # lower legs length
  modeUp: true # True if Z of the platform is higher than Z of base
  PGain: 100
  DGain: 0.1
  jointUpperLimit:
    j1: 3.14
    j2: 3.14
    j3: 3.14
  jointLowerLimit:
    j1: -3.14
    j2: -3.14
    j3: -3.14
  motor1:
    offset: 120 # degree: motor angle at joint = 0 pose
    direction: -1 # is the motor positive direction the same with the joint
  motor2:
    offset: 120 # degree: motor angle at joint = 0 pose
    direction: -1 # is the motor positive direction the same with the joint
  motor3:
    offset: 120 # degree: motor angle at joint = 0 pose
    direction: -1 # is the motor positive direction the same with the joint

ftsensor:
  offset: # will be added to the reading; in tool frame
    fx: 0.577
    fy: 2.27
    fz: -2.37
    tx: 0.235
    ty: -0.117
    tz: 0.00682
  gravity:
    x: -0.00521
    y: 0.0116
    z: -2.28
  COM:
    x: -0.000175
    y: -7.69e-05
    z: -0.0791
  safety:
    fx: 25 # maximum: 20
    fy: 25 # maximum: 20
    fz: 45 # maximum: 60
    tx: 1.2 # maximum: 1
    ty: 1.2 # maximum: 1
    tz: 1.2 # maximum: 1
  transform_sensor_to_tool:
    x: 0 # meter
    y: 0
    z:  0.0962
    qw: 0.9808
    qx: 0
    qy: 0
    qz: 0.1951

##
## Device specific parameters
##

# UR socket
ur:
  portnum: 30003
  ip: "192.168.1.98"
  t: 0.01  # motion time
  lookahead: 0.03  # look ahead time, (0.03 ~ 0.2)
  gain: 2000  # Gain, (100 ~ 2000)


# ati netft
netft:
  ip_address: "192.168.1.1"
  sensor_name: "netft"
  frame_id: "end_effector"
  publish_rate: 1000
  print_flag: false
  file_path: "/home/yifanh/Git/catkin_ws/src/forcecontrol/data/netft_data.txt"

# ABB EGM
egm:
  portnum: 6510
  print_flag: false
  file_path: "/home/yifanh/Git/catkin_ws/src/forcecontrol/data/egm_data.txt"
